packages-content-model/roosterjs-content-model-plugins/lib/edit/keyboardInput.ts

```diff
@@ -44,14 +45,18 @@ export function keyboardInput(editor: IContentModelEditor, rawEvent: KeyboardEve
     }
 }
 
-function shouldInputWithContentModel(selection: DOMSelection | null, rawEvent: KeyboardEvent) {
+function shouldInputWithContentModel(
+    selection: DOMSelection | null,
+    rawEvent: KeyboardEvent,
+    isMac?: boolean
+) {
     if (!selection) {
         return false; // Nothing to delete
     } else if (
         !isModifierKey(rawEvent) &&
         (rawEvent.key == 'Enter' || rawEvent.key == 'Space' || rawEvent.key.length == 1)
     ) {
-        return selection.type != 'range' || !selection.range.collapsed; // TODO: Also handle Enter key even selection is collapsed
+        return selection.type != 'range' || (!selection.range.collapsed && !isMac); // TODO: Also handle Enter key even selection is collapsed
```

Comment:
Thanks for handling this quickly.

Actually, can you try checking editor.isInIME() or rawEvent.isComposing instead? We still need to handle the ENTER key for expanded range when not using IME.

You can refer to here https://github.com/microsoft/roosterjs/blob/236d0afdfb5f94f80151abaa02040a82d49197de/packages-content-model/roosterjs-content-model-core/lib/corePlugin/ContentModelFormatPlugin.ts#L93